include/deal.II/lac/generic_linear_algebra.h

```diff
@@ -68,10 +68,51 @@ namespace LinearAlgebraPETSc
 {
   using namespace dealii;
 
+  /**
+   * Typedef for the base class for preconditioners.
+   */
+  typedef PETScWrappers::PreconditionerBase PreconditionerBase;
+
+  /**
+   * Typedef for the base class for linear solvers.
+   */
+  typedef PETScWrappers::SolverBase SolverBase;
+
+  /**
+   * Typedef for the BiCGStab linear solver.
+   */
+  typedef PETScWrappers::SolverBicgstab SolverBicgstab;
+
+  /**
+   * Typedef for the CG linear solver.
+   */
   typedef PETScWrappers::SolverCG SolverCG;
+
+  /**
+   * Typedef for the CG Squared linear solver.
+   */
+  typedef PETScWrappers::SolverCGS SolverCGS;
+
+#ifdef DEAL_II_PETSC_WITH_MUMPS
+
+  /**
+   * Typedef for the direct linear solver (via MUMPS).
+   */
+  typedef PETScWrappers::SparseDirectMUMPS SolverDirect;
```

Comment:
That seems dangerous if we decide to add another sparse direct solver (such as the SuperLU one requested recetly). I would import it as `SparseDirectMUMPS`, even though that doesn't allow us to have generic names across implementations.